### PR TITLE
MC-38: Add an option to choose if products not handled by Akeneo should be deleted/disabled or not during an export.

### PR DESCRIPTION
**Pull request by damien-carcel**
Bug fix: [no]
Feature addition: [yes]
Backwards compatibility break: [no]
Phpspec specification passes: [no]
Checkstyle issues: [no]
ChangeLog updated: [yes]
Fixes the following issue:

### CODE REVIEW

File: Cleaner/ProductCleaner.php [hunk 1]
```diff
@@ -191,20 +232,35 @@ protected function handleProductNotInPimAnymore(array $product)
...
     /**
      * Handle product for the given action
-     * @param array  $product
-     * @param string $action
+     * @param array   $product
+     * @param string  $notAnymoreAction
+     * @param boolean $removeProductsNotHandledByPim
      */
-    protected function handleProduct(array $product, $action)
+    protected function handleProduct(array $product, $notAnymoreAction, $removeProductsNotHandledByPim)
```
Comment (fitn) [inline Cleaner/ProductCleaner.php]: Set a default parameter on the last parameter to avoid kind of BC break

File: Cleaner/ProductCleaner.php [hunk 2]
```diff
@@ -191,20 +232,35 @@ protected function handleProductNotInPimAnymore(array $product)
...
     {
-        if ($action === self::DISABLE) {
+
+        if (
+            false === $removeProductsNotHandledByPim &&
+            in_array($product['type'], $this->productTypesNotHandledByPim)
+        ) {
+            $this->stepExecution->incrementSummaryInfo('product_not_removed');
+            $this->addWarning('Non removed product\'s SKU: %sku%', ['%sku%' => $product['sku']], $product);
+            return;
```
Comment (fitn) [inline Cleaner/ProductCleaner.php]: Add a break line before return statement (PHPCS rule)

File: Resources/translations/messages.en.yml [hunk 3]
```diff
@@ -11,7 +11,7 @@ pim_magento_connector:
         simple_product_export.export.title: Simple product export
         configurable_product_export.export.title: Configurable product export
         association_product_export.export.title: Product association export
-        simple_product_clean.clean.title: Simple product clean
+        simple_product_clean.clean.title: Product clean
```
Comment (fitn) [inline Resources/translations/messages.en.yml]: If you change Simple product by Product, you've to change it everywhere

File: Cleaner/ProductCleaner.php [hunk 4]
```diff
@@ -30,6 +30,25 @@ class ProductCleaner extends Cleaner
...
+
+    /**
+     * @var string
+     */
+    protected $notCompleteAnymoreAction;
+
+    /**
+     * @var boolean
+     */
+    protected $removeProductsNotHandledByPim;
```
Comment (willy-ahva) [inline Cleaner/ProductCleaner.php]: Is PMD ok with those var length ?

File: Resources/translations/messages.en.yml (hunk 3 above)
Comment (damien-carcel) [inline Resources/translations/messages.en.yml]: Oups, didn't see that one…

Comment (damien-carcel) [inline Resources/translations/messages.en.yml]: Correction, I saw it ! But in this case, it really is a simple product export, as you also have configurable product and product association export.
In the case of the cleaning, all product types are cleaned.

File: Cleaner/ProductCleaner.php (hunk 4 above)
Comment (damien-carcel) [inline Cleaner/ProductCleaner.php]: Indeed it isn't. I will shorter them.

File: Resources/translations/messages.en.yml (hunk 3 above)
Comment (fitn) [inline Resources/translations/messages.en.yml]: Okip ;)

**Pull request merged**